src/lib.rs

```diff
@@ -1,14 +1,84 @@
-mod totp;
-mod hotp;
+use hmac::{
+    crypto_mac::InvalidKeyLength,
+    Hmac, Mac, NewMac,
+};
+use sha1::Sha1;
 
-pub fn digest() {}
-pub fn generate_secret(){}
-pub fn get_otp_auth_url(){}
+type HmacSha1 = Hmac<Sha1>;
+
+pub mod hotp;
+pub mod totp;
+
+enum HmacFunction<A> {
+    Sha1(A),
+}
+
+pub enum Algorithm {
+    Sha1,
+    // Todo (Jayhelton)Implement other hash functions
+    // Sha256,
+    // Sha512,
+}
+
+// pub enum Encoding {
+//     Ascii,
+//     Hex,
+//     Base32,
+//     Base64,
+// }
+
+pub fn digest(
+    secret: String,
+    counter: u128,
+    algorithm: Algorithm,
+) -> std::result::Result<Vec<u8>, InvalidKeyLength> {
```

Comment:
This all looks great! The only thing I would consider would be to explicitly `use std::result::{Result}` since there are multiple Results in the core libs, `thread` for instance, and you're using `Result` on L54 too